@Test
public
void
constructorTest
(
)
throws
Exception
{
CtAbstractInvocation
<
?
>
ctAbstractInvocation
=
this
.
getInvocationFromMethod
(
"testConstructor"
)
;
Assert
.
assertTrue
(
ctAbstractInvocation
instanceof
CtConstructorCall
<
?
>
)
;
CtExecutableReference
<
?
>
executableReference
=
ctAbstractInvocation
.
getExecutable
(
)
;
Assert
.
assertNotNull
(
executableReference
)
;
Constructor
<
?
>
constructor
=
executableReference
.
getActualConstructor
(
)
;
Assert
.
assertNotNull
(
constructor
)
;
Assert
.
assertEquals
(
"Hello World"
,
constructor
.
newInstance
(
(
(
CtLiteral
<
?
>
)
ctAbstractInvocation
.
getArguments
(
)
.
get
(
0
)
)
.
getValue
(
)
)
)
;
}
